Components/DayCalendar.js

```diff
@@ -9,50 +12,251 @@ var {
   ScrollView,
   View,
   Text,
+  PanResponder,
+  Image,
 } = React;
 
+type Event = {
+  title: string;
+  startAt: Date;
+  length: number;
+};
+
 var HOURS_COUNT = 25;
 var HOURS = [];
 for (var i = 0; i < HOURS_COUNT; i++) {
-    var hour = moment().hour(i).second(0).minute(0).format('h:mma');
+    var hour = moment().hour(i).minute(0).second(0).format('h:mma');
     HOURS.push(hour);
 }
+var MINUTES_INA_DAY = 60*24;
+
+var DayCalendar = React.createClass({
+
+  panResponder: {},
+  resizeResponder: {},
+  draggableViewPreviousTop: 0,
+  draggableViewPreviousHeight: 0,
+  draggableView: (null : ?{ setNativeProps(props: Object): void }),
+  resizeHandleView: (null : ?{ setNativeProps(props: Object): void }),
+  draggableViewTopAndHeight: {},
+  scrollView: (null : ?{ setNativeProps(props: Object): void }),
+  totalHeight: 0,
+  beginOfDay: null,
+  endOfDay: null,
+
+  propTypes: {
+    style: View.propTypes.style,
+    eventBoxStyle: View.propTypes.style,
+    day: React.PropTypes.instanceOf(Date).isRequired, 
+    events: React.PropTypes.array.isRequired,
+    newEvent: React.PropTypes.object,
+    timelineHeight: React.PropTypes.number.isRequired,
+    timelineGap: React.PropTypes.number.isRequired,
+    onNewEventStartAtChange: React.PropTypes.func,
+    onNewEventLengthChange: React.PropTypes.func,
+  },
+
+  getDefaultProps: function() {
+    return { day: new Date(), events: [], timelineHeight: 1, timelineGap: 59 };
```

Comment:
propTypes checks type at runtime and this is their old approach. Now they have **flow** which is compile time type system. In the future, I think they may deprecate propTypes.